__init__.py

```diff
@@ -661,11 +726,23 @@ def handle_stt_selected(self, message):
                     callback=handle_intent_aborted)
     def handle_tts_menu(self):
         self.state = SetupState.SELECTING_TTS
-        self.gui["offlineMale"] = self.setup.offline_male_tts_module
-        self.gui["onlineMale"] = self.setup.online_male_tts_module
-        self.gui["offlineFemale"] = self.setup.offline_female_tts_module
-        self.gui["onlineFemale"] = self.setup.online_female_tts_module
-        self.handle_display_manager("BackendLocalTTS")
+        model = get_tts_lang_configs(lang=self.selected_language, include_dialects=True)
+        supported_tts_engines = list()
+        for engine, configs in model.items():
+            # For Display purposes, we want to show the engine name without the underscore or dash and capitalized all
+            plugin_display_name = engine.replace("_", " ").replace("-", " ").title()
```

Comment:
the configs usually contain a "display_name" key set by the plugin itself, that should be checked and only if not set trigger this logic. While this key is optional in plugins i think all our plugins provide it

Reply:
display_name does not exist for engine name, only the internal configuration entries have a display_name, where is the display_name in this output for example: ``` {'ovos-tts-plugin-server': [{'lang': 'en-us', 'url': 'https://sam.jarbasai.online', 'gender': 'male', 'display_name': 'S.A.M. (jarbasai.online)', 'offline': False}, {'lang': 'en-us', 'url': 'https://pico.jarbasai.online', 'gender': 'female', 'display_name': 'PicoTTS (jarbasai.online)', 'offline': False}, {'lang': 'en-us', 'url': 'https://glados.jarbasai.online', 'gender': 'female', 'display_name': 'Glados (jarbasai.online)', 'offline': False}, {'lang': 'en-us', 'url': 'https://mimic.jarbasai.online', 'gender': 'male', 'display_name': 'Alan Pope [Mimic 1] (jarbasai.online)', 'offline': False}, {'lang': 'en-us', 'url': 'https://r2d2.jarbasai.online', 'gender': 'neutral', 'display_name': 'R2D2 (jarbasai.online)', 'offline': False}], 'ovos-tts-plugin-mimic2': [{'voice': 'kusal', 'gender': 'male', 'display_name': 'Kusal', 'offline': False}, {'voice': 'nancy', 'gender': 'female', 'display_name': 'Nancy', 'offline': False}, {'voice': 'ljspeech', 'gender': 'female', 'display_name': 'LJSpeech', 'offline': False}]} ```

there is only the key "ovos-tts-plugin-server" from which it builds the engine display_name

Reply:
plugin_display_name is not 'display_name': 'S.A.M. (jarbasai.online)' for example in GUI its "OVOS TTS Plugin Server", the display_name here in the above json output in configs that is 'display_name': 'S.A.M. (jarbasai.online)' is the voice name not plugin_display_name

Reply:
ok, see comment above, in TTS you are using both names for voice and for plugin, so this comment does not apply here

in STT section above you should do just like in `supported_tts_engines` below and include both display names then

Reply:
both are shown
![tts-list](https://user-images.githubusercontent.com/19663666/195080760-1d2e0a11-99cd-483b-a6b9-2258f554da87.png)

Reply:
not in STT, only in TTS....